src/Gate.jl

```diff
@@ -206,6 +206,15 @@ abstract type Control{Op<:Operator} <: Operator{NamedTuple{(:target,),Tuple{Oper
 Base.length(::Type{Control{T}}) where {T<:Operator} = 1 + length(T)
 parameters(::Type{Control{Op}}) where {Op} = parameters(Op)
 
+"""
+    SU(N)(lane_1, lane_2, ..., lane_log2(N))
+
+A multi-qubit random unitary operator that acts on `log2(N)` qubits.
+"""
+abstract type SU{N} <: Operator{NamedTuple{(:values, :N), Tuple{Matrix, Int}}} where {T, N<:Int} end
+Base.length(::Type{SU{N}}) where {N} =
+    (N & (N - 1) == 0 && N > 0) ? log2(N) |> Int : throw(DomainError(N, "N must be a power of 2"))
+
```

Comment:
Didn't we agreed that `N` was the number of qubits?
`log2(N)` can carry problems like... what if you create `SU{3}`? This won't work.

Reply:
In general, I have seen that `SU{2}` is a one-qubit gate and `SU{4}` a two-qubit gate. `SU{3}` would throw an error on the constructor.